Improve docs in os::windows::ffi and os::windows::fs

excaliburHisSheath opened the pull request:
Part of #29367

This PR makes changes to the documentation in `os::windows::ffi` and `os::windows::fs` with the goal of fleshing them out and bringing them in line with Rust's quality standards.

r? @steveklabnik

rust-highfive commented:
Thanks for the pull request, and welcome! The Rust team is excited to review your changes, and you should hear from @steveklabnik (or someone else) soon.

If any changes to this PR are deemed necessary, please add them as extra commits. This ensures that the reviewer can see what has changed since they last reviewed the code. Due to the way GitHub handles out-of-date commits, this should also make it reasonably obvious what issues have or haven't been addressed. Large or tricky changes may require several passes of review and changes.

Please see [the contribution instructions](https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md) for more information.

TimNN commented:
Hi @excaliburHisSheath and thanks for the PR! We'll periodically check in on it to make sure that @steveklabnik or someone else from the team reviews it soon.

Mark-Simulacrum commented:
@steveklabnik Just a reminder that this is still looking for a review from you.

cc @frewsxcv

frewsxcv commented on src/libstd/sys/windows/ext/ffi.rs:
```diff
@@ -42,11 +56,29 @@ impl OsStringExt for OsString {
+    /// Re-encodes an `OsStr` as a wide character sequence, i.e. potentially
+    /// ill-formed UTF-16.
+    ///
+    /// This is lossless: calling [`OsString::from_wide()`] and then
```
very minor, but can you remove all the `()` parens attached to method/function names in these docs? for the time being, the current policy is to remove them in documentation. see also https://github.com/rust-lang/rust/pull/40456

excaliburHisSheath replied:
Ah, I was under the impression that the `()` parens were the desired style, thanks for clarifying that this isn't the case (I can update documentation for my own crates as well).

excaliburHisSheath replied:
For my own curiosity, is there some discussion around why the decision was made to remove the `()` parens when referencing function in documentation? Personally I liked that style, so I'm curious as to why the change was made.

steveklabnik replied:
It wasn't that a change was made; it was that while proposing https://github.com/rust-lang/rfcs/blob/master/text/1574-more-api-documentation-conventions.md, we suggested that style, but it didn't make it through the process.

(I really like the `()`s so I was just adding them in before we had an actual convention)

frewsxcv commented on src/libstd/sys/windows/ext/mod.rs:
```diff
@@ -8,11 +8,13 @@
+//! Provides access to platform-level information for Windows, and exposes
+//! Windows-specific idioms that would otherwise be inappropriate as part
+//! the core `std` library. These extensions allow developers to use
+//! `std` types and idioms with Windows in a way that the noraml
```
typo: 'noraml'

frewsxcv commented on src/libstd/sys/windows/ext/fs.rs:
````diff
@@ -197,42 +270,157 @@ impl OpenOptionsExt for OpenOptions {
+    ///
+    /// # Examples
+    ///
+    /// ```ignore
````
this example (and potentially other examples) could be `no_run` instead of `ignore` so they get compiled, but not run

excaliburHisSheath replied:
Cool, I wasn't aware of the distinction. Is there a general guideline to when examples should be `no_run` vs `ignore` vs tested normally?

frewsxcv replied:
https://doc.rust-lang.org/book/documentation.html#running-documentation-tests

steveklabnik replied:
the short answer is, if it doesn't compile, `ignore`, and if it compiles but you don't want it to run (say for example, a network test), then `no_run`.

excaliburHisSheath commented:
@frewsxcv I believe I've addressed your concerns. I've left a number of tests `no_run` since they'll create a file `foo.txt` and it seems like bad practice to have tests create random files on disk. If it's preferred that we run the tests anyway I can change them, I think the tests still pass when run.

I've also made some changes to make the code style for the various `OpenOptionsExt` examples more consistent.

frewsxcv commented:
Looks great, thanks!

@bors r+ rollup

bors commented:
:pushpin: Commit 4cd838b has been approved by `frewsxcv`

<!-- @bors r=frewsxcv 4cd838b274db73be8c974e5335b28f1b6f6b75a0 -->

bors commented:
:hourglass: Testing commit 4cd838b with merge 8d5a9ac...

excaliburHisSheath commented:
@frewsxcv Heads up, I pushed another commit after you approved because I noticed a doc test was failing. It doesn't look like the automated tests have caught the failure yet, though. I don't know if you need to do anything to make sure the new commit gets included in the merge.

frewsxcv commented:
@bors r+ rollup

bors commented:
:pushpin: Commit a892925 has been approved by `frewsxcv`

<!-- @bors r=frewsxcv a89292514b6d4c4e9be7d6d527ad72e08e27173c -->
